kyaml/setters2/settersutil/fieldsetter.go

```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 	}
 
 	// Update the resources with the new value
-	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath}
+	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath, NoDeleteFiles: true}
 	s := &setters2.Set{Name: fs.Name}
 	err := kio.Pipeline{
 		Inputs:  []kio.Reader{inout},
-		Filters: []kio.Filter{kio.FilterAll(s)},
+		Filters: []kio.Filter{FilterAll(s)},
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 	return s.Count, err
 }
+
+// FilterAll applies the set filter for all yaml nodes and only returns the nodes whose
+// corresponding file has at least one node with input setter
+func FilterAll(s *setters2.Set) kio.Filter {
```

Comment:
make this private, no reason to export it

Reply:
Done